### ISSUE TITLE
shp2geobuf fails with shapefile containing null geometries

### ISSUE THREAD

**Issue by BergWerkGIS** [mapbox/geobuf#55]
```
node node_modules\geobuf\bin\shp2geobuf contour_5.shp > contour_5.geobuf

node_modules\geobuf\encode.js:51
    if (obj.type === 'FeatureCollection') {
           ^
TypeError: Cannot read property 'type' of null
    at analyze (node_modules\geobuf\encode.js:51:12)
    at analyze (node_modules\geobuf\encode.js:56:9)
    at analyze (node_modules\geobuf\encode.js:52:51)
    at encode (node_modules\geobuf\encode.js:26:5)
    at node_modules\geobuf\bin\shp2geobuf:9:26
    at node_modules\geobuf\node_modules\shapefile\index.js:15:5
    at node_modules\geobuf\node_modules\shapefile\read.js:27:11
    at notify (node_modules\geobuf\node_modules\shapefile\node_modules\queue-async\queue.js:47:18)
    at node_modules\geobuf\node_modules\shapefile\node_modules\queue-async\queue.js:39:16
    at FSReqWrap.oncomplete (fs.js:95:15)
```

Logging `obj` at the [start of `analyze` function](https://github.com/mapbox/geobuf/blob/ba044600d25aa7dda3c669e975d98af32354e1ad/encode.js#L45) shows this before the TypeError occurs:

```
-------obj:
 { type: 'Feature',
  properties: { ID: 51, ELEV: 375 },
  geometry: null }
-------obj:
 null
```

**Comment by jlaine** [comment]
What is the correct representation of 'null' in geobuf, as I don't see a dedicate entry into geobuf.proto? Is it a json_type containing the litteral 'null'?

**Comment by ivorblockley** [comment]
Geojson features containing null geometries can be encountered in real [geojson datasets](https://catalog.data.gov/dataset/city-employee-salaries-1st-quarter-2017-094e3/resource/641e26b8-6e34-4a7f-8f74-d821b215a65f) and are explicitly permitted in [RFC7946](https://tools.ietf.org/html/rfc7946#section-3.2)).

Currently geobuf parsing of geojson features with null geometries fails in the manner described in this issue.

An attempt at allowing representation of null geometries in geobuf is made in #94

**Comment by mourner** [comment]
Fixed in #94
